PU-Adapter/PknodFetcher.cs

```diff
@@ -68,6 +81,52 @@ private static string VerifyAndFormatPersonNumber(string personnummer)
             return personnummer;
         }
 
+        private string FetchFromPUWithCert(string personnummer, string serviceName, X509Certificate2 certificate) 

```

Comment:
Is it really necessary to duplicate most of the code?
Wouldn't it make sense to just check if the `certificate2` property is not null, just like the check for UserName `if (!string.IsNullOrEmpty(UserName))`?